examples/auth-http-basic/lib/basicAuth.ts

```diff
@@ -0,0 +1,50 @@
+import { NextRequest, NextResponse } from "next/server";
+
+interface BasicAuthMiddlewareCredentials {
+  username: string;
+  password: string;
+}
+
+type BasicAuthMiddlewareCheck = (
+  credentials: BasicAuthMiddlewareCredentials
+) => boolean;
+
+const createMiddleware = (check: BasicAuthMiddlewareCheck) => {
+  return (req: NextRequest) => {
+    const basicAuthHeader = req.headers.get("authorization");
+
+    if (basicAuthHeader) {
+      const credentials = parseHeader(basicAuthHeader);
+      const isValid = check(credentials);
+
+      if (isValid) {
+        return NextResponse.next();
+      }
+    }
+
+    const requestHeaders = new Headers(req.headers);
+    requestHeaders.set("www-authenticate", 'Basic realm="Site"');
+
+    const url = `${req.nextUrl.protocol}//${req.nextUrl.host}/401`;
+    return NextResponse.rewrite(url, {
+      status: 401,
+      headers: requestHeaders,
+    });
+  };
+};
+
+const parseHeader = (headerVal: string) => {
+  const authValue = headerVal.split(" ")[1];
+  const [username, password] = Buffer.from(authValue, "base64")
```

Comment:
This works fine locally but when deploying to Cloud Run using a `node:18-alpine` Docker image it was telling me `from` wasn't a function. Alternative is to use the deprecated [atob](https://developer.mozilla.org/en-US/docs/Web/API/atob) which does indeed work.

I assume this is to do with the edge middleware runtime, but this confuses me because:
1) I would have expected for this to fail locally (like other edge errors appear)
2) Cloud run is running as I said using a `node:18` image.

<img width="1313" alt="Screenshot 2023-04-22 at 17 46 20" src="https://user-images.githubusercontent.com/1992355/233796648-70fa33d8-f038-463a-bbc8-19d2ce64c6c7.png">

Any insights here would be appreciated :) 

Please let me know if I should update this.